### PR TITLE
feat: handle tsv + center crop

### PR DESCRIPTION
**Pull request by borisdayma**
This PR adds the following:
* handle tsv files
* allow to resize using a center crop method

### CODE REVIEW

File: img2dataset/downloader.py [hunk 1]
```diff
@@ -38,12 +38,14 @@ def resize_image(img_stream, image_size, resize_mode, resize_only_if_bigger):
         original_height = img.shape[0]
         original_width = img.shape[1]
         if not resize_only_if_bigger or img.shape[0] > image_size or img.shape[1] > image_size:
-            if resize_mode == "border":
-                img = resize_with_border(img, image_size)
-            elif resize_mode == "no":
+            if resize_mode == "no" or image_size == original_height == original_width:
```
Comment (borisdayma) [inline img2dataset/downloader.py:41]: Avoid resizing if it's not needed

File: img2dataset/downloader.py [hunk 2]
```diff
@@ -56,12 +58,20 @@ def resize_keep_ratio(im, desired_size=256):
     old_size = im.shape[:2] # old_size is in (height, width) format
 
     ratio = float(desired_size)/min(old_size)
-    new_size = tuple([int(x*ratio) for x in old_size])
+    new_size = tuple([int(round(x*ratio)) for x in old_size])
```
Comment (borisdayma) [inline img2dataset/downloader.py:61]: An image with 412px would be resized to 255 instead of 256 because of rounding error.

**Review by borisdayma — COMMENTED** [review]
I added a few comments.
For resizing functions based on opencv I would actually suggest to directly use [albumentations](https://github.com/albumentations-team/albumentations) which is optimized for it (and will provide many convenient options & debugging errors). For example [SmallestMaxSize](https://albumentations.ai/docs/api_reference/augmentations/geometric/resize/#albumentations.augmentations.geometric.resize.SmallestMaxSize) corresponds to `resize_keep_ratio`.

File: img2dataset/downloader.py [hunk 3]
```diff
@@ -237,9 +247,11 @@ def download_one_file(url_list):
             with open(url_list, encoding='utf-8') as file:
                 images_to_dl = [(url, ) for url in file.readlines()]
             column_list=["url"]
-        elif input_format == "csv" or input_format=="parquet":
+        elif input_format in ["csv", "tsv", "parquet"]:
             if input_format == "csv":
                 df = pd.read_csv(url_list)
+            elif input_format == "tsv":
+                df = pd.read_csv(url_list, sep='\t')
```
Comment (Skylion007) [inline img2dataset/downloader.py:254]:
```suggestion
                df = pd.read_table(url_list)
```

**Review by Skylion007 — COMMENTED** [review]
isn't read_table the proper way to read tsv?

File: img2dataset/downloader.py (hunk 3 above)
Comment (borisdayma) [reply img2dataset/downloader.py:254]: I think it's the exact same. Let me know which one you prefer.

Comment (Skylion007) [reply img2dataset/downloader.py:254]: read_table.

Comment (rom1504) [reply img2dataset/downloader.py:254]: Why ? Read csv seems more explicit

**Review by rom1504 — APPROVED** [review]

File: img2dataset/downloader.py (hunk 1 above)
Comment (rom1504) [reply img2dataset/downloader.py:41]: ok

File: img2dataset/downloader.py (hunk 2 above)
Comment (rom1504) [reply img2dataset/downloader.py:61]: ok

**Comment by rom1504** [comment]
thanks!

**Pull request merged**